amazon-kinesis-client/src/main/java/software/amazon/kinesis/processor/RecordProcessorCheckpointer.java

```diff
@@ -145,6 +144,9 @@ void checkpoint(String sequenceNumber, long subSequenceNumber)
     PreparedCheckpointer prepareCheckpoint()
             throws KinesisClientLibDependencyException, InvalidStateException, ThrottlingException, ShutdownException;
 
+    PreparedCheckpointer prepareCheckpoint(byte[] applicationState)
```

Comment:
Let's add doc for all the new methods here

Reply:
Added java docs